### PR TITLE
899 specific site

### LINKED ISSUE #899: Add specific site/provider names to data requirements when a user is logged in.
When I am logged in, the site data definition updates to show the options that are accepted by ECE Reporter:
Screen Shot 2020-12-09 at 1 07 53 PM

### PR DESCRIPTION
**Pull request by mmazanec22**
Closes #899

### CODE REVIEW

File: client/src/components/DataDefinitionsTable/DataDefinitionsTable.tsx [hunk 1]
```diff
@@ -65,7 +68,20 @@ const DataDefinitionsTable: React.FC<DataDefinitionsTableProps> = ({
     );
   }
 
-  const columnMetadataBySection: { [key: string]: ColumnMetadata[] } = {};
+  if (user) {
+    const siteRow = filteredColumnMetadata.find(
+      (row) => row.propertyName === 'site'
+    );
+    siteRow!.columnFormatters = getSiteFormatters(user.sites || []);
```
Comment (bamader) [inline client/src/components/DataDefinitionsTable/DataDefinitionsTable.tsx:75]: Should this `!` be a `?`? If not, what does the exclamation point operator do before a dot (haven't seen that before)?

File: client/src/components/DataDefinitionsTable/DataDefinitionsTable.tsx [hunk 2]
```diff
@@ -65,7 +68,20 @@ const DataDefinitionsTable: React.FC<DataDefinitionsTableProps> = ({
...
+    const providerRow = filteredColumnMetadata.find(
+      (row) => row.propertyName === 'providerName'
+    );
+    providerRow!.columnFormatters = getProviderFormatters(user.organizations);
```
Comment (bamader) [inline client/src/components/DataDefinitionsTable/DataDefinitionsTable.tsx:79]: Same here

File: client/src/components/DataDefinitionsTable/utils.tsx [hunk 3]
```diff
@@ -54,3 +55,55 @@ export const getMarkdownStyledFormatOptionsList = (formatString: string) => {
...
+        </ul>
+      </div>
+    </td>
+  ),
+});
+
+export const getProviderFormatters = (organizations?: Organization[]) => ({
+  [DATA_DEF_COLUMN_NAMES.format]: () => {
+    if (!organizations?.length) {
+      // TODO: is there a case where there is no provider name?
```
Comment (bamader) [inline client/src/components/DataDefinitionsTable/utils.tsx:85]: I don't remember how we set it up, but a single-site user with no org-level permissions and site-level access might not have any organizations in this property array

**Review by bamader — APPROVED** [review]

File: client/src/components/DataDefinitionsTable/utils.tsx (hunk 3 above)
Comment (jlhogan) [reply client/src/components/DataDefinitionsTable/utils.tsx:85]: https://github.com/ctoec/data-collection/blob/base/src/controllers/users.ts#L19
nope, site-specific users will have organizations array populated (but not organizationIds)

**Review by jlhogan — APPROVED** [review]

File: client/src/components/DataDefinitionsTable/DataDefinitionsTable.tsx (hunk 1 above)
Comment (mmazanec22) [reply client/src/components/DataDefinitionsTable/DataDefinitionsTable.tsx:75]: Asserts that it exists: https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-0.html#non-null-assertion-operator

**Pull request merged**